src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -201,47 +220,61 @@ public bool TryReceiveOne(out MessageEnvelope envelope, TimeSpan? max, Cancellat
             return InternalTryReceiveOne(out envelope, max, cancellationToken, true);
         }
 
+        /// <inheritdoc cref="TryReceiveOne(out MessageEnvelope, TimeSpan?, CancellationToken)"/>
+        public async ValueTask<(bool success, MessageEnvelope envelope)> TryReceiveOneAsync(TimeSpan? max, CancellationToken cancellationToken)
+        {
+            return await InternalTryReceiveOneAsync(max, cancellationToken, true).ConfigureAwait(false);
+        }
+
         private bool InternalTryReceiveOne(out MessageEnvelope envelope, TimeSpan? max, CancellationToken cancellationToken, bool shouldLog)
         {
-            bool didTake;
+            var received = InternalTryReceiveOneAsync(max, cancellationToken, shouldLog).GetAwaiter().GetResult();
+            envelope = received.envelope;
+            return received.success;
+        }
+
+        private async ValueTask<(bool success, MessageEnvelope envelope)> InternalTryReceiveOneAsync(TimeSpan? max, CancellationToken cancellationToken, bool shouldLog)
+        {
+            (bool didTake, MessageEnvelope env) take;
             var maxDuration = GetTimeoutOrDefault(max);
-            var start = Now;
+            var start = Now;    
             if (maxDuration.IsZero())
             {
                 ConditionalLog(shouldLog, "Trying to receive message from TestActor queue. Will not wait.");
-                didTake = _testState.Queue.TryTake(out envelope);
+                take = await _testState.Queue.TryTakeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This case requires the test kit to return immediately with a value, regardless of the queue state. Don't use the async version of this method, use `TryTake` instead.